packages/sanes-chrome-extension/src/extension/background/actions/createPersona/requestHandler.ts

```diff
@@ -49,4 +49,22 @@ export class RequestHandler {
 
     return size;
   }
+
+  public static purge(senderUrl: string): void {
+    const initialSize = RequestHandler.instance.length;
+    for (let i = 0; i < initialSize; i++) {
+      const req = RequestHandler.instance[i];
+      if (req.sender !== senderUrl) {
+        continue;
+      }
+
+      // Note here we are mutating the array placing in first positions request to be rejected
+      RequestHandler.instance.splice(i, 1);
+      RequestHandler.instance.splice(0, 0, req);
```

Comment:
Simpler: `RequestHandler.instance.unshift(req)`

Reply:
The thing is, when we reject the request, in the callback it calls to `solve()` which removes the first element.

I believe that was my first approach and it failed, anyway I will give another shoot while @albertandrejev reviews it. I have something in mind.

Reply:
This particular comment is just about readability of the existing logic. `unshift` is like `push` but adds in the front instead of the back